apps/service_broadcast/lib/broadcast/init.ex

```diff
@@ -1,16 +1,22 @@
 defmodule Broadcast.Init do
+  use Annotated.Retry
+
   use Initializer,
     name: __MODULE__,
     supervisor: Broadcast.Stream.Supervisor
 
+  @retry with: constant_backoff(100) |> take(10)
   def on_start(state) do
-    Broadcast.Stream.Store.get_all!()
-    |> Enum.reject(&is_nil/1)
-    |> Enum.reject(&Broadcast.Stream.Store.done?(&1))
-    |> Enum.each(fn load ->
-      Broadcast.Stream.Supervisor.start_child(load)
-    end)
-
-    {:ok, state}
+    store = Broadcast.Stream.Store.get_all!()
+    case(store) do
+      {:error, _} -> {:error, "Broadcast failed to read state from store during startup"}
+      _ ->
+        store |> Enum.reject(&is_nil/1)
+        |> Enum.reject(&Broadcast.Stream.Store.done?(&1))
+        |> Enum.each(fn load ->
+          Broadcast.Stream.Supervisor.start_child(load)
+        end)
+        {:ok, state}
+    end
```

Comment:
```suggestion
    safe_retrieve()
    |> Enum.reject(&is_nil/1)
    |> Enum.reject(&Broadcast.Stream.Store.done?(&1))
    |> Enum.each(fn load ->
           Broadcast.Stream.Supervisor.start_child(load)
         end)

    {:ok, state}
  end
    
  @retry with: constant_backoff(100) |> take(10)
  defp safe_retrieve()
    Broadcast.Stream.Store.get_all!()
  catch
    _, reason ->
      Logger.error(fn ->
        "State tables not ready; retrying..."
      end)
  end
```

Reply:
I'm just seeing this one. Brian suggested moving the Annotated Retry up to the main Initializer. I'll touch base and see if there's a way to combine the two approaches.